src/components/wallet/NewMnemonicConfirm.tsx

```diff
@@ -0,0 +1,214 @@
+import React, { useState } from 'react'
+import { Card, Col, Row, Button, Input, Form } from 'antd'
+import { RedoOutlined } from '@ant-design/icons'
+import { Store } from 'antd/lib/form/interface'
+
+type WordType = {
+  text: string
+  _id: string
+  sequence?: number
+  error?: boolean
+  selected?: boolean
+}
+const mnemonic = 'real debris regret sea auto random agree police uncover gloom cloud ribbon'
+
+const MnemonicConfirmScreen: React.FC = (): JSX.Element => {
+  const [wordsList, setWordsList] = useState<WordType[]>([])
+  const [shuffledWordsList, setShuffledWordsList] = useState<WordType[]>([])
+  const [loadingMsg] = useState<string>('')
+  const [mnemonicError, setMnemonicError] = useState<string>('')
+  const [initialized, setInitialized] = useState<boolean>(false)
+
+  function init() {
+    const words = mnemonic.split(' ')
+    if (words && !initialized) {
+      const res = words.map((e: string) => {
+        const uniqueId = Math.random().toString(36).substring(2) + Date.now().toString(36)
```

Comment:
This function to get a unique id looks unsafe, in a worse case it would return a same id. If you really want to have a unique id use a library such as [uniqid]( https://www.npmjs.com/package/uniqid) for that. Besides that, please extract any helper function from component into a `helpers/xxx` module and add tests. That's not necessary by using external libraries such as `uniqid`.

Reply:
ah ok good point. I did check the math, but your point to use a standard lib is appropriate.